### PR TITLE
Add all post call to get effective routes

### PR DESCRIPTION
**Pull request by zaynchen115**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by msJinLei** [comment]
please fix the following static analysis issue
```
  D:\a\_work\1\s\artifacts//StaticAnalysisResults\HelpIssues.csv Errors
  "Assembly","HelpFile","Target","Severity","ProblemId","Description","Remediation"
  "Az.Network","","Get-AzVHubOutboundRoutes","1","6050","Help missing for cmdlet Get-AzVHubOutboundRoutes implemented by class Microsoft.Azure.Commands.Network.GetAzureRmVHubOutboundRoutesCommand","Add Help record for cmdlet Get-AzVHubOutboundRoutes to help file."
  
  D:\a\_work\1\s\artifacts//StaticAnalysisResults\ExampleIssues.csv Errors
  "Module","Cmdlet","Example","Line","RuleName","ProblemId","Severity","Description","Extent","Remediation"
  "Az.Network","Get-AzVHubInboundRoutes","1","6","Invalid_Parameter_Name","5011","1","Get-AzVHubRouteTable -ResourceId is not a valid parameter name.","-ResourceId","Check validity of the parameter -ResourceId."
  "Az.Network","Get-AzVHubInboundRoutes","1","6","Invalid_Parameter_Name","5011","1","Get-AzVHubRouteTable -VirtualWanResourceType is not a valid parameter name.","-VirtualWanResourceType","Check validity of the parameter -VirtualWanResourceType."
```

File: src/Network/Network/help/Get-AzVHubEffectiveRoutes.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Network.dll-Help.xml
+Module Name: Az.Network
+online version: https://learn.microsoft.com/powershell/module/az.network/get-azvhubeffectiveroutes
+
+schema: 2.0.0
+---
+
+# Get-AzVHubEffectiveRoutes
```
Comment (msJinLei) [inline src/Network/Network/help/Get-AzVHubEffectiveRoutes.md:9]: Please use singular for cmdlet name

File: src/Network/Network/help/Get-AzVHubInboundRoutes.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Network.dll-Help.xml
+Module Name: Az.Network
+online version: https://learn.microsoft.com/powershell/module/az.network/get-azvhubinboundroutes
+
+schema: 2.0.0
+---
+
+# Get-AzVHubInboundRoutes
```
Comment (msJinLei) [inline src/Network/Network/help/Get-AzVHubInboundRoutes.md:9]: Please use singular for cmdlet name

File: src/Network/Network/ChangeLog.md [hunk 3]
```diff
@@ -48,6 +48,10 @@
 * Onboarded `Microsoft.HardwareSecurityModules/cloudHsmClusters` to private link cmdlets
 * Updated cmdlet to add the property of DisableRequestBodyEnforcement, RequestBodyInspectLimitInKB and DisableFileUploadEnforcement 
     - `New-AzApplicationGatewayFirewallPolicySetting`
+* Add new cmdlets to get virtual hub effective routes and in/outbound routes 
+    - `Get-AzVHubEffectiveRoutes`
+    - `Get-AzVHubInboundRoutes`
+    - `Get-AzVHubOutboundRoutes`
```
Comment (msJinLei) [inline src/Network/Network/ChangeLog.md:54]: Please also generate help doc for Get-AzVHubOutboundRoutes

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please add test cases for the new added cmdlets.

**Comment by zaynchen115** [comment]
@msJinLei please help review, all tests added and passed. Lmk if any new branch needs to be merged

**Comment by msJinLei** [comment]
> @msJinLei please help review, all tests added and passed. Lmk if any new branch needs to be merged

@zaynchen115 
You don't address all of the review comments.
We already code freeze.
I recommend you to split your PR.
One part is to make change to an unreleased cmdlet.
Please submit a new PR to just contain this part and target to branch release-2023-05-23.

And the other parts of this PR will be merged to main and target to the next regular release.

**Comment by zaynchen115** [comment]
@msJinLei what's the breaking changes are you mentioning. Can I merge to main?
I thought there's a regression in swagger, maybe I can CP if there's another fix another in PS also?

**Comment by zaynchen115** [comment]
create a PR to target main https://github.com/Azure/azure-powershell/pull/21850

**Comment by msJinLei** [comment]
> @msJinLei what's the breaking changes are you mentioning. Can I merge to main? I thought there's a regression in swagger, maybe I can CP if there's another fix another in PS also?

**Please only keep the changes that relates to GetAzureRmVHubEffectiveRoutesCommand.cs in this PR.**

What you change to this cmdlet is the [breaking change] (https://github.com/Azure/azure-powershell/blob/main/documentation/breaking-changes/breaking-changes-definition.md).  It is a new added cmdlet in this sprint. Currently the cmdlet is not released and so these changes are not identified as the breaking change.

**Comment by msJinLei** [comment]
close due to be duplicate with https://github.com/Azure/azure-powershell/pull/21850

**Pull request closed without merging**